src/cljs/bluegenes/components/ui/constraint.cljs

```diff
@@ -95,74 +97,107 @@
 (defn set-text-value [node value]
   (when node (-> (sel1 node :input) (dommy/set-value! value))))
 
+(def fmt "yyyy-MM-dd")
+(def date-fmt (time-format/formatter fmt))
+
+(defn read-day-change
+  "Convert DayPicker input to the string we use as constraint."
+  [date _mods picker]
+  (let [input (-> (ocall picker :getInput)
+                  (oget :value))]
+    ;; `date` can be nil if it's not a valid date. We use the raw input text in
+    ;; that case, to accomodate alien calendars.
+    (or (some->> date
+                 (time-coerce/from-date)
+                 (time-format/unparse date-fmt))
+        input)))
+
 (defn constraint-text-input
```

Comment:
This function is massive - maybe we could break it into smaller chunks, e.g. by sub-types?

Reply:
Good idea!

Reply:
the fixes are elegant and beautiful 🌷